crates/sui-graphql-rpc/src/types/cursor.rs

```diff
@@ -386,21 +407,26 @@ impl<C: CursorType + Eq + Clone + Send + Sync + 'static> Page<C> {
                 }
 
                 // Page drawn from the front, and the cursor for the first element does not match
-                // `after`. This implies the bound was invalid, so we return an empty result.
-                (Some(a), Some(f), _, _, End::Front) if f != *a => {
+                // `after`. If that cursor is not from a `scan_limit`, then it must have appeared in
+                // the previous page, and should also be at the tip of the current page. This
+                // absence implies the bound was invalid, so we return an empty result.
+                (Some(a), Some(f), _, _, End::Front) if f != *a && !a.is_scan_limited() => {
                     return (false, false, vec![].into_iter());
                 }
 
                 // Similar to above case, but for back of results.
-                (_, _, Some(l), Some(b), End::Back) if l != *b => {
+                (_, _, Some(l), Some(b), End::Back) if l != *b && !b.is_scan_limited() => {
                     return (false, false, vec![].into_iter());
                 }
 
-                // From here onwards, we know that the results are non-empty and if a cursor was
-                // supplied on the end the page is being drawn from, it was found in the results
-                // (implying a page follows in that direction).
-                (after, _, Some(l), before, End::Front) => {
-                    let has_previous_page = after.is_some();
+                // From here onwards, we know that the results are non-empty. In the forward
```

Comment:
yes, this still holds true, since if the results were empty they'd get caught at
```
                (_, None, _, _, _) | (_, _, None, _, _) => {
                    return (false, false, vec![].into_iter());
                }
```